src/test/java/org/folio/rest/impl/EholdingsProvidersImplTest.java

```diff
@@ -423,6 +425,68 @@ public void shouldUpdateOnlyProviderTagsWhenNoPackagesAreSelected() throws IOExc
     }
   }
 
+  @Test
+  public void shouldUpdateTagsOnPutTags() throws IOException, URISyntaxException {
+    try {
+      List<String> newTags = Arrays.asList(STUB_TAG_VALUE, STUB_TAG_VALUE_2);
+      sendPutTags(Arrays.asList(STUB_TAG_VALUE, STUB_TAG_VALUE_2), PROVIDER_TAGS_PATH);
+      List<String> tagsAfterRequest = TagsTestUtil.getTagsForRecordType(vertx, RecordType.PROVIDER);
+      assertThat(tagsAfterRequest, containsInAnyOrder(newTags.toArray()));
+    } finally {
+      TagsTestUtil.clearTags(vertx);
+      TestUtil.clearDataFromTable(vertx, PROVIDERS_TABLE_NAME);
+    }
+  }
+
+  @Test
+  public void shouldUpdateTagsOnPutTagsWithAlreadyExistingTags() throws IOException, URISyntaxException {
+    try {
+      TagsTestUtil.insertTag(vertx, STUB_VENDOR_ID, RecordType.PROVIDER, STUB_TAG_VALUE);
+      List<String> newTags = Arrays.asList(STUB_TAG_VALUE, STUB_TAG_VALUE_2);
+      sendPutTags(Arrays.asList(STUB_TAG_VALUE, STUB_TAG_VALUE_2), PROVIDER_TAGS_PATH);
+      List<String> tagsAfterRequest = TagsTestUtil.getTagsForRecordType(vertx, RecordType.PROVIDER);
+      assertThat(tagsAfterRequest, containsInAnyOrder(newTags.toArray()));
+    } finally {
+      TagsTestUtil.clearTags(vertx);
+      TestUtil.clearDataFromTable(vertx, PROVIDERS_TABLE_NAME);
+    }
+  }
+
+  @Test
+  public void shouldUpdateTagsOnPutTagsAndParams() throws IOException, URISyntaxException {
```

Comment:
because "?param" will be ignored, this test seems to be exactly the same as "shouldUpdateTagsOnPutTagsWithAlreadyExistingTags"